src/containers/shared/components/Streams.jsx

```diff
@@ -97,6 +139,74 @@ class Streams extends Component {
 
     this.mounted = false;
     clearInterval(this.purge);
+    clearInterval(this.purgeAll);
+  }
+
+  // handle ledger messages
+  handleLedger(data) {
+    const ledger = this.addLedger(data);
+    const { ledger_hash: ledgerHash, ledger_index: ledgerIndex, txn_count: txnCount } = data;
+
+    Log.info('new ledger', ledgerIndex);
+    ledger.ledger_hash = ledgerHash;
+    ledger.txn_count = txnCount;
+    ledger.close_time = (data.ledger_time + EPOCH_OFFSET) * 1000;
+
+    const metrics = this.updateMetrics(data.fee_base / 1000000);
+    return {
+      ledger,
+      metrics,
+    };
+  }
+
+  // handle validation messages
+  handleValidation(data) {
+    const { ledger_hash: ledgerHash, validation_public_key: pubkey } = data;
+    const ledgerIndex = Number(data.ledger_index);
+
+    if (!this.isValidatedChain(ledgerIndex)) {
+      return undefined;
+    }
+
+    this.addLedger(data);
+
+    this.setState(prevState => {
+      if (!prevState.allValidators[pubkey]) {
+        const allValidators = {
+          [pubkey]: { pubkey, ledger_index: 0 },
+        };
+        return { allValidators };
+      }
+      return {};
+    });
+
+    const { allValidators } = this.state;
+    if (
+      allValidators[pubkey].ledger_hash !== ledgerHash &&
+      ledgerIndex > allValidators[pubkey].ledger_index
+    ) {
+      this.setState(prevState => {
+        const newValidatorData = Object.assign(prevState.allValidators[pubkey], {
+          ledger_hash: ledgerHash,
+          ledger_index: ledgerIndex,
+          last: Date.now(),
+        });
+        const newAllValidators = Object.assign(prevState.allValidators, {
+          [pubkey]: newValidatorData,
```

Comment:
What does `[pubkey]` syntax mean? (Is that a key that has the value of an array?)

Reply:
It's syntax that essentially means the key will have the value of that variable (without the `[]` it would be a key with the value of the string `"pubkey"`)